RecoParticleFlow/PFProducer/src/MLPFModel.cc

```diff
@@ -0,0 +1,212 @@
+        py = ref->py();
+        pz = ref->pz();
+        eta = ref->eta();
+        phi = ref->phi();
+        energy = ref->pt() * cosh(ref->eta());
+        trajpoint = orig2->indTrajPoint();
+        charge = ref->charge();
+      }
+    } else if (type == reco::PFBlockElement::GSF) {
+      //requires to keep GsfPFRecTracks
+      const auto* orig2 = (const reco::PFBlockElementGsfTrack*)&orig;
+      pt = orig2->Pin().pt();
+      px = orig2->Pin().px();
+      py = orig2->Pin().py();
+      pz = orig2->Pin().pz();
+      eta = orig2->Pin().eta();
+      phi = orig2->Pin().phi();
+      energy = pt * cosh(eta);
+      if (!orig2->GsftrackRefPF().isNull()) {
+        charge = orig2->GsftrackRefPF()->charge();
+      }
+    } else if (type == reco::PFBlockElement::ECAL || type == reco::PFBlockElement::PS1 ||
+               type == reco::PFBlockElement::PS2 || type == reco::PFBlockElement::HCAL ||
+               type == reco::PFBlockElement::HO || type == reco::PFBlockElement::HFHAD ||
+               type == reco::PFBlockElement::HFEM) {
+      const auto& ref = ((const reco::PFBlockElementCluster*)&orig)->clusterRef();
+      if (ref.isNonnull()) {
+        eta = ref->eta();
+        phi = ref->phi();
+        px = ref->position().x();
+        py = ref->position().y();
+        pz = ref->position().z();
+        energy = ref->energy();
+        layer = ref->layer();
+        depth = ref->depth();
+      }
+    } else if (type == reco::PFBlockElement::SC) {
+      const auto& clref = ((const reco::PFBlockElementSuperCluster*)&orig)->superClusterRef();
+      if (clref.isNonnull()) {
+        eta = clref->eta();
+        phi = clref->phi();
+        px = clref->position().x();
+        py = clref->position().y();
+        pz = clref->position().z();
+        energy = clref->energy();
+      }
+    }
+
+    float typ_idx = static_cast<float>(elem_type_encoding.at(orig.type()));
+
+    //Must be the same order as in tf_model.py
+    return std::array<float, NUM_ELEMENT_FEATURES>({{typ_idx,
+                                                     pt,
+                                                     eta,
+                                                     phi,
+                                                     energy,
+                                                     layer,
+                                                     depth,
+                                                     charge,
+                                                     trajpoint,
+                                                     eta_ecal,
+                                                     phi_ecal,
+                                                     eta_hcal,
+                                                     phi_hcal,
+                                                     muon_dt_hits,
+                                                     muon_csc_hits}});
+  }
+
+  float normalize(float in) {
+    if (std::abs(in) > 1e4) {
```

Comment:
```suggestion
    if (std::abs(in) > 1e4f) {
```
to avoid conversions to doubles

Reply:
fixed in 3c0bf883ee2